lib/Parse/ParseStmt.cpp

```diff
@@ -1808,10 +1808,8 @@ ParserResult<Stmt> Parser::parseStmtRepeat(LabeledStmtInfo labelInfo) {
   SourceLoc whileLoc;
 
   if (!consumeIf(tok::kw_while, whileLoc)) {
-    diagnose(whileLoc, diag::expected_while_after_repeat_body);
-    if (body.isNonNull())
-      return body;
-    return makeParserError();
+    diagnose(body.getPtrOrNull()->getEndLoc(), diag::expected_while_after_repeat_body);
```

Comment:
@DougGregor  Yep, but look up a couple lines. There's already a check for body.isNull() which creates a synthetic body (lines 1804-1806), so body can never be null here. statements.swift:269 tests exactly this case.